Feature/search-screen-UI

baraa0abd opened the pull request:
# Feature: Search Screen UI

This branch implements the UI for the search screen in the Movio app.

## Key Features and Changes

- **Search Screen UI:**
  - Added a new `SearchScreen` composable for searching functionality.
  - Integrated a `SearchHeader` for user input and search actions.
  - Displayed recent search history with the ability to add, remove, and clear items.
  - Used a `RecentSearchList` component to show and manage recent searches.
  - Managed state with a `SearchViewModel` using LiveData and coroutines.
  - Handled loading and error states for recent searches.

---

This work enhances the user experience by providing a responsive and interactive search interface, making it easier for users to manage and revisit their recent searches.

baraa0abd committed:
feat: Implement Search Screen and Recent Searches functionality

This commit introduces a new SearchScreen composable and a SearchViewModel to manage search-related state and logic.

Key features added:
- Search input field with clear and submit actions.
- Display of recent searches.
- Ability to add, remove, and clear recent search items.
- Filtering of recent searches based on current input.
- Basic loading and error states for recent searches.

New components:
- `SearchHeader`: Composable for the search input field and title.
- `RecentSearchList`: Composable to display the list of recent search queries.
- `RecentSearchItem`: Data class representing a single recent search.

Drawable resources for icons (clock and delete) have also been added.
Gradle configuration has been updated, including SDK versions and dependencies for Compose, ViewModel, LiveData, and Retrofit.
The `MainActivity` has been updated to display the `SearchScreen`.

baraa0abd committed:
chore: Update README.md with new content

fatimafaisalghazi commented on README.md:
the readme file should contains the descrption about the entier project not just one feature

fatimafaisalghazi commented on app/src/main/java/com/example/movio/MainActivity.kt:
```diff
@@ -4,17 +4,14 @@ import android.os.Bundle
-            }
-
+                SearchScreen()
         }
```
why do you push a change on mainActivity

baraa0abd committed:
refactor: Update RecentSearchList and SearchHeader to use AppTheme for styling

This commit refactors the RecentSearchList and SearchHeader components to utilize the AppTheme for consistent styling. Changes include:
- Replaced hardcoded padding, colors, and font styles with values from AppTheme.
- Improved layout and spacing for better visual consistency across the app.
- Enhanced readability by using theme-defined text styles and colors.

baraa0abd committed:
fix: update text styles and remove unused code

This commit updates the text styles in RecentSearchList and SearchHeader components to use the new text style definitions. It also removes unused code in MainActivity and RecentSearchList. Additionally, the project's JDK language level has been updated to 17, and the designSystem module has been added as a dependency.

baraa0abd committed:
chore: remove README.md file

karrar-abbas commented on app/src/main/java/com/example/movio/component/SearchHeader.kt:
```diff
@@ -0,0 +1,90 @@
+                    onSearch = { onSubmit() }
+                )
+            )
+            IconButton(onClick = onClear, modifier = Modifier.size(AppTheme.spacing.xLarge)) {
```
you use material 3 components; use component in the design system module instead

karrar-abbas commented on app/src/main/java/com/example/movio/component/RecentSearchList.kt:
```diff
@@ -0,0 +1,129 @@
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Icon
```
Use design system module components

Mousaber00 commented on .idea/gradle.xml:
```diff
@@ -6,7 +6,7 @@
         <option name="testRunner" value="CHOOSE_PER_TEST" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
-        <option name="gradleJvm" value="#GRADLE_LOCAL_JAVA_HOME" />
+        <option name="gradleJvm" value="17" />
```
not supposed to be committed with changes i guess

Mousaber00 commented on app/src/main/java/com/example/movio/component/RecentSearchList.kt:
```diff
@@ -0,0 +1,129 @@
+                .fillMaxWidth()
+                .padding(vertical = AppTheme.spacing.small)
+        ) {
+            // Header
```
unnecessary comment

Mousaber00 commented on app/src/main/java/com/example/movio/component/SearchHeader.kt:
```diff
@@ -0,0 +1,90 @@
+            TextField(
+                value = searchQuery,
+                onValueChange = onSearchChange,
+                placeholder = { Text("Search", color = AppTheme.colors.surfaceColor.onSurfaceVariant) },
```
refactor code

YaseenOmar commented on app/src/main/java/com/example/movio/component/RecentSearchList.kt:
```diff
@@ -0,0 +1,129 @@
+                verticalAlignment = Alignment.CenterVertically
+            ) {
+                Text(
+                    text = "Recent Search",
```
hardcoding , use strings file

baraa0abd committed:
Refactor: Use DesignSystem components in RecentSearchList and SearchHeader

This commit replaces standard Material Design components (Icon, IconButton, Text) with their DesignSystem equivalents (MovioIcon, MovioButton, MovioText) in the `RecentSearchList` and `SearchHeader` composables.

It also updates these components to use string resources for text and content descriptions, improving internationalization and accessibility.

baraa0abd committed:
Refactor: Integrate ViewModel with SearchScreen

This commit integrates `SearchListViewModel` into the `SearchScreen` composable.

Key changes:
- `SearchScreen` now instantiates and uses `SearchListViewModel`.
- `SearchListViewModel` manages the state of recent searches, including loading, adding, removing, and clearing them.
- `SearchScreen` observes the `recentSearches` flow from the ViewModel and updates the UI accordingly.
- Error handling for adding recent searches has been added.

baraa0abd committed:
feat: add RecentSearchItem data class and RecentSearchRepository interface

This commit introduces a data class `RecentSearchItem` to represent a recent search query with its ID, query string, timestamp, and an optional search count.

It also defines a `RecentSearchRepository` interface with methods for retrieving, adding, removing, and clearing recent search items.

baraa0abd committed:
feat: implement SearchListViewModel

This commit introduces `SearchListViewModel`, which manages the state and logic for recent searches.

It includes functionality to:
- Load recent searches
- Add a new recent search
- Remove a specific recent search
- Clear all recent searches

The ViewModel utilizes `InMemoryRecentSearchRepository` for data operations and exposes recent searches as a `StateFlow`.

baraa0abd committed:
refactor home and build gridle

baraa0abd committed:
delete uncomplete test file

Mohammed-qmr committed:
Add SearchScreen with ViewModel integration via Koin, search functionality, recommended & explore sections, and updated movie card with online image support

baraa0abd committed:
Merge branch 'develop' into feature/search-screen-ui

# Conflicts:
#	app/build.gradle.kts

baraa0abd committed:
Refactor: Use DesignSystem components in RecentSearchList and SearchHeader

This commit replaces standard Material Design components (Icon, IconButton, Text) with their DesignSystem equivalents (MovioIcon, MovioButton, MovioText) in the `RecentSearchList` and `SearchHeader` composables.

It also updates these components to use string resources for text and content descriptions, improving internationalization and accessibility.

Additionally, the `SearchHeader` component now uses `BasicTextField` instead of `TextField` for more customization and control over its appearance and behavior.

The `RecentSearchList` and `SearchHeader` composables have been moved from the `app` module to the `presentation` module, along with the `searchListScreen` composable and its associated ViewModel.

baraa0abd committed:
feat: add Room dependencies and data module to presentation

This commit introduces the Room persistence library by adding its dependencies to the `gradle/libs.versions.toml` file.

Additionally, the `data` module is now included as an implementation dependency in the `presentation/build.gradle.kts` file. This allows the presentation layer to access data layer components, such as repositories.

Finally, an import statement for `InMemoryRecentSearchRepository` was reordered in `SearchListViewModel.kt`.

baraa0abd committed:
feat: implement in-memory recent search repository and view model

This commit introduces an in-memory implementation of the `RecentSearchRepository` interface and a `SearchListViewModel` to manage recent search data.

The `InMemoryRecentSearchRepository` provides basic CRUD operations for recent search items, storing them in a mutable list.

The `SearchListViewModel` utilizes this repository to expose recent searches as a `StateFlow`, allowing UI components to observe changes and update accordingly. It also provides methods for adding, removing, and clearing recent searches.

baraa0abd committed:
style: Remove SearchScreen and update theme

This commit removes the SearchScreen from MainActivity and updates the theme to use AppTheme instead of MovioTheme.

baraa0abd committed:
style: Remove SearchScreen and update theme

This commit removes the SearchScreen from MainActivity and updates the theme to use AppTheme instead of MovioTheme.

baraa0abd committed:
Merge branch 'feature/search-screen-ui' of https://github.com/Washington-D-C-Squad/Movio into feature/search-screen-ui

# Conflicts:
#	app/src/main/java/com/example/movio/MainActivity.kt

baraa0abd committed:
Refactor: Enhance SearchScreen with DesignSystem components and styling

This commit refactors the `ViewSearchScreen.kt` to utilize components from the `DesignSystem` module.

Key changes include:
- Replaced `OutlinedTextField` with a custom search bar using `BasicTextField` and `MovioIcon`, styled according to `AppTheme`.
- Replaced `Text` with `MovioText` for consistent typography.
- Replaced `CircularProgressIndicator` with a custom `MovioIcon` for the loading state.
- Updated text and icon colors to align with `AppTheme`.
- Added Coil and Koin compose dependencies to `build.gradle.kts`.

baraa0abd committed:
chore: remove AppModule.kt

This commit removes the `AppModule.kt` file, which was previously used for Koin dependency injection.

Mohamedragabali committed:
Merge branch 'develop' into feature/search-screen-ui

# Conflicts:
#	app/build.gradle.kts
#	domain/src/main/java/com/example/domain/RecentSearchItem.kt
#	domain/src/main/java/com/example/domain/RecentSearchRepository.kt
#	gradle/libs.versions.toml
#	presentation/build.gradle.kts
#	presentation/src/main/java/com/example/presentation/component/movioCards/VerticalMovioCard.kt

Mohamedragabali committed:
refactor branch

Mohamedragabali committed:
refactor code
